docs/guide/env-and-mode.md

```diff
@@ -108,6 +108,8 @@ Vite also supports replacing env variables in HTML files. Any properties in `imp
 
 If the env doesn't exist in `import.meta.env`, e.g. `%NON_EXISTENT%`, it will be ignored and not replaced, unlike `import.meta.env.NON_EXISTENT` in JS where it's replaced as `undefined`.
 
+Given that Vite is used by many frameworks, it is purposedly unopinionated about more complex replacements like conditionals. Vite can be extended using [an existent userland plugin](https://github.com/vitejs/awesome-vite#transformers) or a custom plugin that implements the [`transformIndexHtml` hook](./api-plugin#transformindexhtml).
```

Comment:
```suggestion
Given that Vite is used by many frameworks, it is intentionally unopinionated about complex replacements like conditionals. Vite can be extended using [an existing userland plugin](https://github.com/vitejs/awesome-vite#transformers) or a custom plugin that implements the [`transformIndexHtml` hook](./api-plugin#transformindexhtml).
```

`purposedly` (doesn't have the `d`) but I figured to use "intentionally" since "purposely" is often used to describe someone instead.